FightPandemics/Features/Post/CreatePost/OfferRequest/OfferRequest.swift

```diff
@@ -0,0 +1,135 @@
+//
+//  OfferRequest.swift
+//  FightPandemics
+//
+//  Created by ehsan sat on 5/16/20.
+//
+//  Copyright (c) 2020 FightPandemics
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+import UIKit
+
+class OfferRequest: UIView {
+    private var topLine = UIView()
+    private var questionLbl = UILabel()
+    private var offerBtn = UIButton()
+    private var reqBtn = UIButton()
+    private var closeBtn = UIButton()
+    private var confirmBtn = UIButton()
+    private var firstVSV = UIStackView()
+    private var firstHSV = UIStackView()
+    private var finalVSV = UIStackView()
+    func setUp () {
+        setUpUpper()
+        setUpLower()
+        setUpConstraints()
+    }
+    func setUpUpper () {
+        topLine.backgroundColor = UIColor(hexString: "#F3F4FE")
+        topLine.frame.size.width = 116
+        topLine.frame.size.height = 4
+        topLine.clipsToBounds = true
+        topLine.layer.masksToBounds = true
+        topLine.layer.cornerRadius = topLine.frame.size.height / 2
+        self.addSubview(topLine)
+        questionLbl.text = "OfferRequestTitle".localized
+        questionLbl.font = Fonts.poppinsBold.customFont(size: 16)
+        questionLbl.textAlignment = .center
+        questionLbl.numberOfLines = 0
+        self.addSubview(questionLbl)
+    }
+    func setUpLower () {
+        offerBtn.setAttributedTitle(NSAttributedString(string: "OfferRequestOfferBtn".localized, attributes: [NSAttributedString.Key.font: Fonts.dmSansRegular.customFont(size: 12), NSAttributedString.Key.foregroundColor: UIColor.black]), for: .normal)
+        offerBtn.setTitleColor(UIColor.white, for: .selected)
+        offerBtn.frame.size.height = 45
+        offerBtn.frame.size.width = UIScreen.main.bounds.width - 46
+        offerBtn.backgroundColor = UIColor(hexString: "#F2F2F2")
+        offerBtn.clipsToBounds = true
```

Comment:
Good job so far, I think it would be great if we can create a UIButton extension with method to handle giving the button corner radius and reduce this customization to one line of code.

Reply:
They are different types of button with different customization. Like in this page, there are two types of button which needed different corner radius and border width and background ..

Reply:
Good, I am fine then on the corner radius then 👍🏻.